code_eval.py

```diff
@@ -89,16 +89,22 @@ def python3(cmd):
 		output = 'unknown error'
 	reply(cmd, output)
 
+def _no_back_ticks(str):
+	return str.strip('`').strip()
+
 def prep_input(args):
 	args = args.lstrip()
-	if args.startswith('```') and args.endswith('```'):
-		split = args[3:].split('\n', 1)
-		if len(split) == 2:
-			language, other_lines = args[3:].split('\n', 1)
-			if language in ['python', 'py', 'javascript', 'js', 'ruby', 'rb']:
-				return other_lines[:-3]
-			# otherwise, after backticks was code, not language: '```print(1)```'
-	return args.strip('`').strip()
+	if args.startswith('```'):
+		code = args[3:].split('```')[0] # ignore anything after the ending ``` (aka discord pings -- @raylu#1337)
+		try:
+			language, other_lines = code.split('\n', 1)
+			if language.rstrip() in ['python', 'py', 'javascript', 'js', 'ruby', 'rb']:
+				return other_lines
+				# otherwise, after backticks was code, not language: '```print(1)```'
+		except ValueError:
```

Comment:
why did you replace the `if len(split) == 2` with the `try`/`except`?

Reply:
If a new line character is not included (ex: ```print(1)```), then an exception is raised.

```
args = '```print(1)```';
code = args.split('```')[1]
language, other_lines = code.split('\n')
print(language)
print(other_lines)

// Exception is thrown 
Traceback (most recent call last):
  File "main.py", line 4, in <module>
    language, other_lines = code.split('\n')
ValueError: not enough values to unpack (expected 2, got 1)
```

This is a result of splitting the initial check.
```py 
if args.startswith('```') and args.endswith('```'):` 

became 

if args.startswith('```'):
		code = args[3:].split('```')[0] # ignore anything after the ending \`\`\` (aka discord pings -- @raylu#1337)
```

Reply:
right. the old code handled this by splitting into 2 parts
```py
split = args[3:].split('\n', 1)
````
and then checking
```py
if len(split) == 2:
```
is there a reason you can't do this too?

Reply:
tests added.